Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -0,0 +1,287 @@
+Then do the same for Site 2 but put in the domain for Site 1 as the "live" one.
+
+At this point you should be able to go to the backoffice of either environment and do a Content transfer to live, and it should end up on the other (Assuming no errors from your custom connector).
+
+## Debugging a transfer
+
+At this point we haven't done anything to the ValueConverter yet, other than return the original value. Now we will attach Visual Studio to the IIS processes and try a transfer to see what it sends along.
+
+* Go to Visual Studio
+* Hit "Attach to Process" (default ALT + CTRL + P)
+* Choose your two IIS processes
+* Add breakpoints in the `ToArtifact` and `FromArtifact` methods
+* Go to the backoffice in Site 1
+* Try to do a content transfer to live (Site 2).
+
+It will hit your breakpoint, and if you continue you will then get an error. On the breakpoint you can see why the error occurs. It should look like this:
+
+![Hitting the breakpoint](images/hitting-breakpoints.png)
+
+Here you can see that value is null, and if you try to return `value.ToString()` you will get a null exception. 
+
+We will change the `ToArtifact` method a little:
+
+```csharp
+public string ToArtifact(object value, PropertyType propertyType, ICollection<ArtifactDependency> dependencies)
+{
+    var svalue = value as string;
+    if (string.IsNullOrWhiteSpace(svalue))
+        return null;
+
+    return value.ToString();
+}
+```
+
+At this point you can't reattach the process as the code has changed. So, build the project, go to `site1/bin` and copy `projectalias.Core.dll` and `projectalias.Core.pdb`. Paste these files into `site2/bin`, attach to the two IIS processes and try another transfer.
+
+The workflow here is not optimal, but a lot quicker than trying to deploy to Cloud everytime, and with this you can attach the debugger as well to help you out.
+
+After copying the dll and pdb files over we are synced up, now attach the debugger and attempt another transfer. Now you will see that `value` is null a few times, then your hardcoded ID a few times, but nothing breaks here. Eventually you will hit the `FromArtifact` method instead:
+
+![Hitting FromArtifact](images/fromArtifact.png)
+
+Here you will notice that the value is what you had returned in `ToArtifact`.
+
+## Creating our ValueConnector
+
+You may have realised at this point that the flow is something like this:
+
+Site 1 content transfer initated -> Hit the `ToArtifact` method on the environment -> Send to Site 2 -> Hit the `FromArtifact` method on Site 2 -> Property data on Site 2
+
+So in our case, what we want to do is to ensure the ID is changed in a transfer. We do this by converting the ID to a GUID in the `ToArtifact` method on Site 1, which will then get transfered to Site 2. On site 2 we will convert it back to an ID in the `FromArtifact` method. This way the user will still see an ID on the content node, but the ID they see will be updated to the correct one.
```

Comment:
Would suggest something like:
"Site 1 content transfer initiated -> Property data is fetched on Site 1 -> Hit the `ToArtifact` method on the environment -> Send to Site 2 -> Hit the `FromArtifact` method on Site 2 -> Property data is saved on Site 2"

and:
"So in our case, what we want to do is to ensure the ID from Site 1 is changed during a transfer to match what the new ID is in Site 2. We do this by converting the ID to a GUID in the `ToArtifact` method on Site 1, which will then get transfered to Site 2. On site 2 we will convert it back to an ID in the `FromArtifact` method. This way the user will still see an ID on the content node, but the ID they see will be updated to the correct one."